### DISCUSSION TITLE
Impact of string names

### DISCUSSION THREAD

**Discussion by sstroemer** [calliope-project/calliope#477] [General]
Preface: A lot of this could be irrelevant (or just plain wrong if I misunderstood something) if it changes in v0.7, please just point that out and ignore the respective points.

All questions talking about a "model" are related to the `national_scale` example model, with `subset_time: ['2005-01-01', '2005-07-01']`, using `calliope_version: 0.6.10`.

---

I've seen that the string names of variables/constraints are pretty verbose (and therefore long), which makes sense for readability / debugging. Is there a way to configure this? Is there a benchmark of how much that impacts build time as well as "pass to solver" time? One possibility (if not already present) would be to keep an internal compression dictionary that encodes frequently used string parts and translates between internal names - that can be read - and the ones written to LP files.

**Comment by brynpickering** [comment]
@sstroemer it would be good to split each of the top-level questions into their own discussion. Will make it easier to respond!

**Reply by sstroemer** [reply]
Edited this one, and split off each question into separate discussions :+1:

**Comment by brynpickering** [comment]
If you use `run.save_logs` when running your model (and maybe _any_ time you ask the calliope model to save to LP - I can't remember exactly), then we trigger a Pyomo option to pass the verbose names to the LP generation. By default, there are no strings passed on and everything is just a number corresponding to the position of the object in Pyomo's register of component objects.

In v0.7, I've made this more explicit by allowing you to give everything verbose string names after you have built the backend model, if you like. I found it takes ~10% of the model build time to stringify everything, although I don't know how that impacts LP build time / footprint on disk.
